### DISCUSSION TITLE
primitive.readyPromise.otherwise is not a function 这个错误怎么解决？

### DISCUSSION THREAD

**Discussion by j524275214** [zouyaoji/vue-cesium#182] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/40748734/161881872-f01aecd2-3dda-4598-91dc-58d2a7aa2fe5.png)
使用chrome打开https://zouyaoji.top/vue-cesium-v2/#/zh/geometry/vc-instance-geometry
官方文档都有问题啊！

**Comment by j524275214** [comment, marked as answer]
在node_modules中找到
![image](https://user-images.githubusercontent.com/40748734/161912676-74c02ddc-0ece-4e88-a017-47ccc13ad9c4.png)
然后
![image](https://user-images.githubusercontent.com/40748734/161912721-05021801-714b-44f0-bab5-fe53b2199635.png)
即可

**Comment by zouyaoji** [comment]
这是因为最新的CesiumJS去掉了when.js，目前已经兼容了，请安装最新的版本。

**Reply by j524275214** [reply]
感谢，不过最新版本需要使用vue3.x，这和许多老的vue2.x项目是不兼容的，因此用上面的办法即可。

**Reply by zouyaoji** [reply]
vue2.x 也兼容解决了这个问题的哈 2.3.5 版本。

**Reply by j524275214** [reply]
![](https://user-images.githubusercontent.com/40748734/163328738-bea5188a-3094-47e7-9386-3a714e507fb0.png)
![](https://user-images.githubusercontent.com/40748734/163328803-e1b1420f-9292-4c4e-bdb2-9b17145cdebb.png)
重新npm i了 但好像还是不行 是我哪里操作有误么

**Reply by zouyaoji** [reply]
不太清楚，删了 node_modules 试试看看吧。

**Reply by j524275214** [reply]
OK 谢谢
